src/Bicep.Core/Registry/Oci/OciManifest.cs

```diff
@@ -29,5 +32,10 @@ public OciManifest(int schemaVersion, string? artifactType, OciDescriptor config
         /// Additional information provided through arbitrary metadata.
         /// </summary>
         public ImmutableDictionary<string, string> Annotations { get; }
+
+        internal static OciManifest? FromBinaryData(BinaryData d)
+        {
+            return JsonConvert.DeserializeObject<OciManifest>(d.ToString());
```

Comment:
We should avoid using  `Newtonsoft.Json` and use `System.Text.Json` instead when the code is not related to ARM template processing.

Reply:
After talking it through with @shenglol we agreed to handle this as a separate PR so it doesn't block this one.

I will create a task for the team to prioritize given it's a bit complex to use the System.Text.JSON package